phpstan.dist.neon

```diff
@@ -5,22 +5,56 @@ includes:
     - dev-tools/phpstan/baseline.php
 
 parameters:
-    level: 6
+    level: 7
     paths:
         - src
         - tests
     excludePaths:
         - tests/Fixtures
+    polluteScopeWithLoopInitialAssignments: true # Do not enforce assignments outside of the loops
     reportUnmatchedIgnoredErrors: true # Do not allow outdated errors in the baseline
     treatPhpDocTypesAsCertain: false
     ignoreErrors:
         - '/^Class [a-zA-Z\\]+ extends @final class PhpCsFixer\\(ConfigurationException\\InvalidConfigurationException|ConfigurationException\\InvalidFixerConfigurationException|Tokenizer\\Tokens|Console\\Command\\FixCommand)\.$/'
         - '/^\$this\(PhpCsFixer\\Tokenizer\\Tokens\) does not accept PhpCsFixer\\Tokenizer\\Token\|null\.$/'
 
+        # We often need to iterate multiple times within single method and we re-use variable name
+        -
+            message: '/^For loop initial assignment overwrites variable \$(i|index|endIndex)+\.$/'
+            paths:
+                - src/Fixer/Comment/CommentToPhpdocFixer.php
+                - src/Fixer/ControlStructure/YodaStyleFixer.php
+                - src/Fixer/Import/GlobalNamespaceImportFixer.php
+                - src/Fixer/Import/OrderedImportsFixer.php
+                - src/Fixer/LanguageConstruct/SingleSpaceAroundConstructFixer.php
+                - src/Fixer/Operator/NoUselessConcatOperatorFixer.php
+                - src/Fixer/PhpUnit/PhpUnitDedicateAssertInternalTypeFixer.php
+                - src/Fixer/ReturnNotation/ReturnAssignmentFixer.php
+                - src/Fixer/Whitespace/StatementIndentationFixer.php
+                - tests/DocBlock/TypeExpressionTest.php
+        -
+            message: '/^Foreach overwrites \$index with its key variable.$/'
+            paths:
+                - src/Fixer/DoctrineAnnotation/DoctrineAnnotationIndentationFixer.php
+                - src/Fixer/Import/GlobalNamespaceImportFixer.php
+                - src/Fixer/LanguageConstruct/GetClassToClassKeywordFixer.php
+
+        # Types related to tokens collections are purposefully narrowed, because we expect only tokens there
```

Comment:
i wonder if we can instruct on class level that item is given type and not `mixed`